### DISCUSSION TITLE
New PCB design (rev B)

### DISCUSSION THREAD

**Discussion by markadev** [markadev/AppleII-VGA#19] [Show and tell]
👋 I finally finished the layout and got some custom PCBs made (from JLCPCB) and they turned out pretty good & function perfectly.  I call this "rev B" and they are committed into the repo. There are some small changes from the original circuit, described in the [Changelog](https://github.com/markadev/AppleII-VGA/blob/main/AppleVGA/CHANGELOG.txt), but there are cuttable jumpers to revert back to the original "rev A" functionality if needed.  The firmware never made use of the "rev A" functionality that I disabled so nothing's really lost.

Thanks to @swetland for sharing [their board design](https://github.com/markadev/AppleII-VGA/discussions/15).  I liked the ideas of being able to reconfigure back to the "rev A" behavior, and using a small PCB to make a neat back-panel connector.

JLCPCB options that I changed from the default value:
 * PCB thickness: 1.2mm
 * Surface Finish: ENIG
 * Remove Order Number: Specify a location
 * Gold Fingers: yes
 * 30°finger chamfered: yes

The design is still geared towards hand soldering & part flexibility:
 * All required parts are through-hole
 * You may use a surface-mount MOSFET instead of the Schottkey diode in the power supply circuit. The MOSFET is more efficient but you may have an easier time finding & soldering the diode.
 * You can surface mount the Pico to avoid the need for pin headers
 * For the DB15HD (VGA) connector you could build a connector board & ribbon cable, or decide to build a cable that connects directly to the 2x4 pin header

![front](https://github.com/markadev/AppleII-VGA/raw/main/docs/board_rev_b.jpg)
![connector](https://github.com/markadev/AppleII-VGA/raw/main/docs/connector_board.jpg)
![installed](https://github.com/markadev/AppleII-VGA/raw/main/docs/installed_in_iie.jpg)
Note: _Slot 4 is now blocked by the connector board in my IIe. It's not a problem for me but something to keep in mind_

**Comment by swetland** [comment]
Nice!  The through-hole components also have the benefit of looking more era-appropriate than the SOICs and 0603s I used.

**Comment by plaes** [comment]
I have also built one of the @swetland's beta boards from July and it is working great.
One of the "itchy" things with the current board designs is that the location of VGA output is blocking other cards. I really like how this is handled on the Apple's serial card:
![image](https://github.com/markadev/AppleII-VGA/assets/91895/d21d51e2-8f84-40c3-b33b-e757d0f81f30)

**Reply by omigeot** [reply]
wouldn't some 90° IDC connectors (like this : 
![image](https://github.com/markadev/AppleII-VGA/assets/576124/b62e92fe-8657-431c-8aad-97d28f265a9c)
) do the trick?

**Reply by markadev** [reply]
Hmm, maybe 90 degree IDCs would work a little better than straight.  Right now if I put my card in slot 5 with disk controller card in slot 6, the top of the ribbon cable connector touches the disk controller card just slightly.  If I removed the extra strain relief clip (and had the cable coming out the other way from the connector) then I don't think clearance would be an issue at all.

This is one of the reasons I switched to using the 2x4 connector, so that I could use a ribbon cable like this😀 

If you didn't see @plaes, Brian switched out the straight pin header for a right-angle pin header (see https://github.com/markadev/AppleII-VGA/discussions/15#discussioncomment-6956615) and the installation looks much more compact.

**Reply by omigeot** [reply]
Another option would be like Retrotink version does : VGA connector on the front end of the card, and then (optionnaly) a regular VGA extension to reach the back of the enclosure.

https://user-images.githubusercontent.com/121696513/248597347-b7b65cbc-0930-425f-b416-7a6d84813e2b.jpg

**Reply by WarlockD** [reply]
Huh.  Maybe this explains why ribbon cables came on the top of the expansion cards.

**Reply by swetland** [reply]
I found that with the appropriate orientation of the VGA connector in the back panel, you can avoid getting in the way of the adjacent card slot.  It's not perfect, but it gets the job done...

![appleiivga-installed](https://github.com/markadev/AppleII-VGA/assets/94628/833e1cd2-52e1-4c7b-8242-6c1ad3690553)

**Comment by Khaled-Alsan** [comment]
It looks great!

**Comment by markadev** [comment]
I did a little reorganization in my IIe so that I could fit a Mockingboard into slot 4.  With a small piece of sheet aluminum I mounted the VGA connector in one of the DB-19 holes.  (So many DB-19 holes and nothing else to put in them!).  That keeps the VGA connector way off to the side of the slots and also keeps enough room for a right-angle adapter I use for the Disk interface.  It's not as elegant as originally had, but this is another solution for freeing up a blocked slot.

![rear](https://github.com/markadev/AppleII-VGA/assets/17986295/3e754a80-cc3e-4b28-ac11-a4ba481b60bd)
![inside](https://github.com/markadev/AppleII-VGA/assets/17986295/894ccbb3-8df7-4404-917f-9dedb8406f8c)

**Comment by plaes** [comment]
Some time ago I complained about the VGA connector blocking the card itself. So I did finally scratch my itch and [designed a two-in-one PCB](https://github.com/plaes/AppleII-VGA/tree/plaes-smd-mod) where connector can be snapped off thanks to mousebites. Board-to-board connections use 8-pin IDC sockets (one of them 90degrees):

![image](https://github.com/markadev/AppleII-VGA/assets/91895/b8154f1c-006c-48e3-8262-386441b32b89)
 ![image](https://github.com/markadev/AppleII-VGA/assets/91895/8a5d6ad0-df97-47f1-9449-82b953db8f9e)
![image](https://github.com/markadev/AppleII-VGA/assets/91895/075a9027-46a8-44ed-8631-0c78a396d8d6)
![image](https://github.com/markadev/AppleII-VGA/assets/91895/3b0d2937-70a4-4410-b3ca-3fe0723f2cf2)


Branch is here: https://github.com/plaes/AppleII-VGA/tree/plaes-smd-mod

**Reply by markadev** [reply]
Neat design! Thanks for sharing!

**Reply by vitasam** [reply]
Hi @plaes, nice design! Is your PCB fully compatible with the original one by firmware?

Btw, will JLCPCB complain that this is actually two boards and charge price correspondingly?

**Reply by plaes** [reply]
> Hi @plaes, nice design! Is your PCB fully compatible with the original one by firmware?

Works for me on Apple IIe, though I haven't tested with latest firmwares.

> Btw, will JLCPCB complain that this is actually two boards and charge price correspondingly?

JLCPCB didn't, but PCBGOGO did.

**Comment by vitasam** [comment]
Wow, I haven't been here in ages - you've started making PCB boards!
I've recently resumed my slow-paced work on my [MPF-II-VGA-proto-board](https://github.com/markadev/AppleII-VGA/discussions/4#discussioncomment-5428226) and realized that manually soldering all those wires is a real hassle.
Could you kindly advise on the best approach? Should I:

(i) Order your PCB board as-is, then cut the gap in the bus connector and make all modifications on the board using a scalpel and soldering jumpers?

or

(ii) Invest time in KiCAD to modify the board design according to my specific needs?
I'd appreciate your input on which option would be more efficient and effective

**Reply by vitasam** [reply]
Or (iii): I will make an intermediate adapter board: MPF-II bus <-> Apple-II bus (maybe with some jumpers to enable/disable/swap lines). This adapter board will let the VGA-board be installed vertically, not horizontally like it is now.

In this case your board design will remain mostly the same, the only changes that came to my mind:

1) VGA connector should be on the board
2) If there is a free GPIO available, a jumper to switch the firmware between "Apple-II" and "MPF-II" modes could be added

**Reply by markadev** [reply]
Hey there! It's up to you but if I were going to go through the trouble of ordering a custom PCB then I would make it as close to the final circuit as possible, maybe even with extra pads for reworking areas I'm not 100% confident in.  Though I'm familiar with PCB design and KiCad already so I'm a little biased 😀 

Building an adapter board is also a very good idea. You could then just purchase a prebuilt VGA card or just a single PCB on eBay if you'd like and then flash on any firmware changes you'd like to make for the MPF-II.  There are a couple board variations that have the VGA connector on the card itself and as long as their instructions link back to this github repo for firmware then they should be suitable.

**Reply by vitasam** [reply]
Hi! Thanks, the more I'm thinking about it, the more I tend to go with option (iii) - make an adapter board. I have some experience with JLCPCB ordering - once it was completely my own KiCad design of a calculator replica, and another time I ordered a third-party board.

And yes - there is no need for additional jumpers on the VGA board - it is easy to flash a new firmware.
Do you know if it's possible to find modern Apple II bus connectors so one can incorporate them into the PCB adapter design?

**Reply by markadev** [reply]
Cool. I believe [EDAC part number 395-050-520-201](https://www.mouser.com/ProductDetail/EDAC/395-050-520-201?qs=U4pz39agNJDEIfodVv79Sg%3D%3D) is a modern equivalent. It's a standard card edge connector so you may be able to find a cheaper off-brand too.

**Reply by vitasam** [reply]
Ah, indeed, the standard connector. I will try to find something.

You said about eBay options for VGA cards. Could you kindly send a link to suitable one that have VGA connector on the card and compatible with your Apple-II FW?

**Reply by markadev** [reply]
Here is [one example](https://www.ebay.com/itm/296108805236).

**Reply by vitasam** [reply]
Option (iii) is getting closer:
![image](https://github.com/user-attachments/assets/f892a331-3deb-4506-b90d-dedcf9b1b50e)

**Comment by vitasam** [comment]
Ok, just ordered the adapter board, let's see

![image](https://github.com/user-attachments/assets/5e703ebd-ce74-48ff-8e94-a07b2f9e0c36)

**Comment by vitasam** [comment]
@markadev do you have some free empty PCB of the VGA-card for sale? Don't want to pay for 5 boards from JLCPCB
